app/scripts/controllers/onboarding.ts

```diff
@@ -0,0 +1,103 @@
+import { ObservableStore } from '@metamask/obs-store';
+import log from 'loglevel';
+
+type flowTypes = 'create' | 'import';
+/**
+ * The state of the OnboardingController
+ */
+export type OnboardingControllerState = {
+  seedPhraseBackedUp: boolean | null;
+  firstTimeFlowType: flowTypes | null;
+  completedOnboarding: boolean;
+  onboardingTabs?: Record<string, string>;
+};
+
+const defaultTransientState: Record<string, Record<string, string>> = {
+  onboardingTabs: {},
+};
+
+const defaultState: Partial<OnboardingControllerState> = {
+  seedPhraseBackedUp: null,
+  firstTimeFlowType: null,
+  completedOnboarding: false,
+};
```

Comment:
How would you feel about using the `satisfies` keyword here instead of a type annotation to a `Partial` type?

```suggestion
const defaultState = {
  seedPhraseBackedUp: null,
  firstTimeFlowType: null,
  completedOnboarding: false,
} satisfies OnboardingControllerState;
```

- `Partial` is inaccurate since we don't want `defaultState`'s properties to be optional.
- On the advantages of `satisfies` over `:`:
  - [Example](https://github.com/MetaMask/contributor-docs/blob/240206-typescript-guidelines-any-as/docs/typescript.md#example-e9b0d703-032d-428b-a232-f5aa56a94470)
  - [Explanation](https://github.com/MetaMask/contributor-docs/blob/240206-typescript-guidelines-any-as/docs/typescript.md#use-the-satisfies-operator-to-enforce-a-type-constraint-or-to-validate-the-assigned-type)